project/server/app.js

```diff
@@ -1,26 +1,155 @@
 const express = require('express');
 const bodyParser = require('body-parser');
+const cors = require('cors');
+const Validation = require('../src/components/validationComponent');
+const mongoose = require('mongoose');
+const MongoClient = require('mongodb');
+const Schema = mongoose.Schema;
+const users = require('./routes/users');
+const router = express.Router();
+const Users = require('./models/Users.js');
+const multer = require('multer');
+const assert = require('assert');
+const jwt = require('jsonwebtoken');
+const {mongoConnect} = require('../src/constants');
+const {storage} = require('./storage');
 
-const db = require('./utils/dataBaseUtils.js');
 
-db.setUpConnection();
+const db = mongoose.connection;
+mongoose.connect(mongoConnect, ((err)=>{
+    if (err) throw err;
+    console.log('Successfully connected');
+}))
 
 const app = express();
+app.use(cors());
+
+app.use('/users', users);
 
 app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
 
-app.get('/notes', (req, res)=>{
-  db.listNotes().then(data => res.send(data));
-});
 
-app.post('/notes', (req, res)=>{
-  db.createNote(req, body).then(data => res.send(data));
-});
+const upload = multer({storage: storage});
 
-app.delete('/notes/:id', (req, res)=>{
-  db.createNote(req, params, id).then(data => res.send(data));
-});
+const generatePsd = (()=>{
+    return Math.random().toString(36).slice(-8);
+})
+
+
+app.post('/addUser', upload.single('photo'), (req, res, next) => {
+    const user = req.body;
+    const psw = generatePsd();
+    try {
+        Users.findOne({email: user.email }, (err, user)=>{
+            if(user){
+                console.log(req.body);
+                return res.send(200, 'emailError');
+            }
+                let nameValid = Validation.validateName(req.body.name);
```

Comment:
Create an additional function which would get the parameters and return the result, are these parameters valid, or not.